strum_macros/src/helpers/metadata.rs

```diff
@@ -39,6 +39,7 @@ pub mod kw {
     custom_keyword!(default_with);
     custom_keyword!(props);
     custom_keyword!(ascii_case_insensitive);
+    custom_keyword!(flatten);
```

Comment:
honestly, the biggest concern I have here is how should `#[strum(flatten)]` interact with other derives

Reply:
https://github.com/Peternator7/strum/pull/369#issuecomment-2254264961

basically this thing